### PR TITLE
FTP - attribute enrichment of FTPFile #153

### PR DESCRIPTION
**Pull request by svezfaz**
Handles #153

### CODE REVIEW

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala [hunk 1]
```diff
@@ -90,9 +115,7 @@ trait CommonFtpSourceSpec extends BaseSpec {
       probe.expectComplete()
 
       val expectedNumOfBytes = getLoremIpsum.getBytes().length * numOfFiles
-      val total = result.foldLeft(0L) {
-        case (acc, next) => acc + next.count
-      }
+      val total = result.map(_.count).sum
```
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: this traverses `result` twice, right?

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpOperations.scala [hunk 2]
```diff
@@ -4,11 +4,13 @@
 package akka.stream.alpakka.ftp
 package impl
 
-import org.apache.commons.net.ftp.{FTP, FTPClient}
+import org.apache.commons.net.ftp.{FTP, FTPClient, FTPFile}
+
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpOperations.scala]: blank line

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala [hunk 3]
```diff
@@ -4,12 +4,17 @@
 package akka.stream.alpakka.ftp
 package impl
 
+import java.io.InputStream
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermissions
+
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala]: blank line

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala [hunk 4]
```diff
@@ -4,12 +4,17 @@
...
 import akka.stream.alpakka.ftp.RemoteFileSettings.SftpSettings
 import com.jcraft.jsch.{ChannelSftp, JSch}
+
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala]: blank line

File: ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala [hunk 5]
```diff
@@ -4,12 +4,17 @@
...
 import scala.collection.immutable
 import scala.util.Try
-import java.io.InputStream
-import java.nio.file.Paths
+
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/impl/SftpOperations.scala]: blank line

File: ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala [hunk 6]
```diff
@@ -4,8 +4,10 @@
 package akka.stream.alpakka.ftp
 
 import akka.stream.alpakka.ftp.FtpCredentials.AnonFtpCredentials
+
```
Comment (juanjovazquez) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala]: blank line

File: ftp/src/test/java/akka/stream/alpakka/ftp/FtpBaseSupport.java [hunk 7]
```diff
@@ -7,10 +7,12 @@
 import akka.stream.ActorMaterializer;
 import akka.stream.Materializer;
 import org.apache.mina.util.AvailablePortFinder;
+
```
Comment (juanjovazquez) [inline ftp/src/test/java/akka/stream/alpakka/ftp/FtpBaseSupport.java]: blank line

File: ftp/src/test/scala/akka/stream/alpakka/ftp/BaseSpec.scala [hunk 8]
```diff
@@ -8,7 +8,8 @@ import akka.stream.IOResult
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
 import org.scalatest.concurrent.ScalaFutures
-import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, Matchers, WordSpecLike}
+import org.scalatest._
+
```
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/BaseSpec.scala]: I'd rather not go with global imports: `org.scalatest._`

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala [hunk 9]
```diff
@@ -3,10 +3,14 @@
  */
 package akka.stream.alpakka.ftp
 
+import java.nio.file.attribute.PosixFilePermission
```
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: So far, I've been putting the java imports at the end of the import list. Just by convention.

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala [hunk 10]
```diff
@@ -3,10 +3,14 @@
...
+
 import akka.stream.IOResult
 import akka.stream.scaladsl.{Keep, Sink}
 import akka.stream.testkit.scaladsl.TestSink
 import org.scalatest.time.{Millis, Seconds, Span}
+
```
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: blank line

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala [hunk 11]
```diff
@@ -3,10 +3,14 @@
...
+import scala.concurrent.duration._
```
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: I'd get rid of the global import on `scala.concurrent.duration._`

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala (hunk 1 above)
Comment (svezfaz) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: Given this is a test, I appreciate increased readability over performance. I am not too adamant about it, though :) I can put it back if there is a strong argument

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala (hunk 11 above)
Comment (svezfaz) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: this is to have the `.millis` and `.minute` conversions from `duration` package object. would you prefer not to use them?

**Comment by svezfaz** [comment]
@juanjovazquez Thanks for reviewing!

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala (hunk 1 above)
Comment (juanjovazquez) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: No strong argument at all. I guess it's just a matter of style.

File: ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala (hunk 6 above)
Comment (ktoso) [inline ftp/src/main/scala/akka/stream/alpakka/ftp/model.scala]: which is ok ;) space between akka deps and language deps is something some people like; I don't mind

File: ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala (hunk 1 above)
Comment (ktoso) [inline ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpSourceSpec.scala]: in a test readability definitely wins over pers in things like these IMHO, thanks!

**Comment by ktoso** [comment]
LGTM, resolved conflicts

**Pull request merged**